### PR TITLE
Update the error in the image block to use snackbars

### PR DESCRIPTION
**Pull request by scruffian**
## Description
Now we have snackbar messages available we can use this in the image block for notifying users when their upload fails.

## How has this been tested?
Try uploading an SVG to the image block. You should see a snackbar message telling you that it didn't work.

## Screenshots 
Screenshot 2019-12-06 at 14 56 15

## Types of changes
UI Improvement

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

**Comment by talldan** [comment]
I'm not convinced this is the right pattern for the image uploads, as for a fairly critical error the warning doesn't seem obvious enough. I think the snackbar works best for simple messages that are quick to read.

**Comment by karmatosed** [comment]
I think we need to take a step back and decide what type of messages we have in which situation. For example, we currently have 'by the block' errors for other issues like gallery and other blocks. If we work on a methodical plan here then we might end up with snackbars here, but we also might not.

I will note the use of snackbars tends to more 'single notice, closing' over having to know contents of error as @talldan points out.

cc @ItsJonQ to look at this from a system view.

**Comment by ItsJonQ** [comment]
@karmatosed Halloo!! 👋 

@scruffian Thank you for making this update! Handling error states/edge-cases with UI interactions is so important.

Re: Snackbar interactions.

Using the Snackbar component to handle errors seems standard, example: 

via https://material.io/components/snackbars/#

Whether it's the **best**/**desired** pattern, that I'm unsure of. For "oh no, something went wrong" type notifications, I found Snackbars to be too **subtle for desktop UI**. This is often because they appear much farther away from the interaction that triggered it.

The alternative UI to show for errors would be a [dismissable alert banner](https://getbootstrap.com/docs/4.0/components/alerts/#dismissing), which is appropriately disruptive to let the user know something went wrong.

@karmatosed correct me if I'm wrong, but I don't think we have that global alert-like pattern in Gutenberg yet. The Snackbar appears to be the closest one. The catch-all for alert handling. (good and bad).

---

TLDR: I vote that this is okay for now.

It uses an existing pattern/component, and users can benefit from this notice until we establish improved patterns for escalating error handling.

**Comment by karmatosed** [comment]
@ItsJonQ right now we use the default WordPress error styling by the block. I think if there is a will to create a snackbar type error it should be used throughout.

**Comment by karmatosed** [comment]
For context,  adding what we have now as message options (including existing patterns):

image

**Comment by shaunandrews** [comment]
> I found Snackbars to be too subtle for desktop UI. This is often because they appear much farther away from the interaction that triggered it.

I can't emphasize this enough. Material is super focused on mobile devices, often at the detriment of larger screens.

My opinion is that adopting the Snackbar pattern in web-native Gutenberg/Core is likely a mistake overall.

**Comment by ItsJonQ** [comment]
Thanks for the thoughts and context @karmatosed + @shaunandrews !

In that case, since the [inline messaging alert](https://user-images.githubusercontent.com/253067/70446367-e3f22280-1a94-11ea-804b-c6f7bf55b793.png) exists, using it would probably be better.

As @shaunandrews had mentioned, having the prompt be localized to the block gives much better awareness of what went wrong.

**Comment by scruffian** [comment]
Seems like we have consensus, thanks all :)
